packages/enzyme-utilities/src/index.ts

```diff
@@ -33,19 +35,23 @@ export function trigger(wrapper: AnyWrapper, keypath: string, ...args: any[]) {
 
   let returnValue: any;
 
-  act(() => {
+  const promise = act(() => {
     returnValue = callback(...args);
-  });
 
-  updateRoot(wrapper);
+    if (isPromise(returnValue)) {
+      return (returnValue as unknown) as Promise<void>;
+    }
+  });
 
-  if (returnValue instanceof Promise) {
-    return returnValue.then(ret => {
+  if (isPromise(returnValue)) {
+    return Promise.resolve(promise as Promise<any>).then(ret => {
```

Comment:
Should leave a comment of why you check that `returnValue` is a promise, not `promise`